qdev_wrappers/T3/customised_instruments.py

```diff
@@ -227,78 +227,66 @@ def __init__(self, name, address, config, **kwargs):
-
-        # Take voltage divider of source/drain into account:
-        dcbias_i = int(config.get('Channel Parameters',
-                                  'source channel'))
-        dcbias = self.channels[dcbias_i].volt
-        self.dcbias = VoltageDivider(dcbias,
-                                     float(config.get('Gain Settings',
-                                                      'dc factor')))
-        self.dcbias.label = config.get('Decadac Channel Labels', dcbias_i)
-
-        # Assign custom variable names
-        lcut = config.get('Channel Parameters', 'left cutter')
-        self.lcut = self.channels[int(lcut)].volt
-
-        rcut = config.get('Channel Parameters', 'right cutter')
-        self.rcut = self.channels[int(rcut)].volt
-
-        jj = config.get('Channel Parameters', 'central cutter')
-        self.jj = self.channels[int(jj)].volt
-
-        rplg = config.get('Channel Parameters', 'right plunger')
-        self.rplg = self.channels[int(rplg)].volt
-
-        lplg = config.get('Channel Parameters', 'left plunger')
-        self.lplg = self.channels[int(lplg)].volt
-
-        self.add_parameter('cutters',
-                           label='{} cutters'.format(self.name),
-                           # use lambda for late binding
-                           get_cmd=self.get_cutters,
-                           set_cmd=self.set_cutters,
-                           unit='V',
-                           get_parser=float)
-
-        # Set up voltage and ramp safetly limits in software
-        ranges = config.get('Decadac Channel Limits')
-        ramp_settings = config.get('Decadac Channel Ramp Setttings')
 
+        '''
+        config file redesigned to have all channels for overview. Indices in config_settings[] for each channel are:
+        0: Channels name for deca.{}
+        1: Channel label
+        2: Channels unit (included as we are using decadac to control the magnet)
+        3: Voltage division factor
+        4: step size
+        5: delay
+        6: max value
+        7: min value
+        8: Fine or coarse mode channel
+        '''
+
+        # Couldnt get this to work with normal parameters so ended quite ugly using 'exec' to define input strings as methods
+        config_file = config.get('Decadac')
+        
         for chan in range(20):
-            try:
-                chan_range = ranges[str(chan)]
-            except KeyError:
-                continue
-            range_minmax = chan_range.split(" ")
-            if len(range_minmax) != 2:
-                raise ValueError(
-                    "Expected: min max. Got {}".format(chan_range))
-            else:
-                rangemin = float(range_minmax[0])
-                rangemax = float(range_minmax[1])
+            config_settings = config_file[str(chan)].split(",")
+            
+            step = float(config_settings[4])
+            delay = float(config_settings[5])
+            self.channels[chan].volt.set_step(step)
```

Comment:
Why do you only set the step for the volt and not for the fine_volt?

Reply:
my thinking was that fine_volt refers back to volt. So by setting volt on all channels I will also define step on the fine_volt implicitly.